pkg/apb/ext_creds.go

```diff
@@ -10,82 +10,68 @@ import (
 	logging "github.com/op/go-logging"
 )
 
-var stillWaitingError = "status: still waiting to start"
 var timeoutFreq = 6    // Seconds
 var totalTimeout = 900 // 15min
 
 // ExtractCredentials - Extract credentials from pod in a certain namespace.
-// HACK ALERT!
-// A lot of the current approach to extracting credentials and monitoring
-// output is *very* experimental and error prone. Entire approach is going
-// to be thrown out and redone asap.
 func ExtractCredentials(
 	podname string, namespace string, log *logging.Logger,
 ) (*ExtractedCredentials, error) {
-	log.Debug("Calling monitorOutput on " + podname)
-	credOut, _ := monitorOutput(podname, namespace)
-	log.Debug("oc log output: %s", string(credOut))
+	credOut := make(chan []byte)
 
-	var creds *ExtractedCredentials
-	var err error
-	creds, err = buildExtractedCredentials(credOut)
+	log.Debug("Calling monitorOutput on " + podname)
+	go monitorOutput(namespace, podname, credOut, log)
```

Comment:
+1 at this time and the way we do the work I don't see the need for a go routine to monitor the output.